frontend/packages/kubevirt-plugin/integration-tests/tests/vm.actions.scenario.ts

```diff
@@ -0,0 +1,334 @@
+      await browser.wait(
+        waitForStringInElement(vmDetailNode(vm.namespace, vm.name), sourceNode),
+        VM_MIGRATION_TIMEOUT_SECS,
+      );
+    },
+    VM_ACTIONS_TIMEOUT_SECS,
+  );
+});
+
+describe('Add/remove disks and NICs on respective VM pages', () => {
+  const testVm = getVmManifest('Container', testName, `vm-disk-nic-${testName}`);
+  const vm = new VirtualMachine(testVm.metadata);
+
+  beforeAll(async () => {
+    createResources([multusNad, testVm]);
+    await vm.action('Start');
+  }, VM_BOOTUP_TIMEOUT_SECS);
+
+  afterAll(() => {
+    deleteResources([multusNad, testVm]);
+  });
+
+  it(
+    'Add/remove disk on VM disks page',
+    async () => {
+      await vm.addDisk(hddDisk);
+      expect(await vm.getAttachedDisks()).toContain(hddDisk);
+
+      let vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(false);
+
+      await vm.action('Restart');
+
+      vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(true);
+
+      await vm.removeDisk(hddDisk.name);
+      expect(await vm.getAttachedDisks()).not.toContain(hddDisk);
+
+      await vm.action('Restart');
+
+      vmi = await vm.navigateToVmi(TABS.OVERVIEW);
+      expect((await vmi.getVolumes()).includes(hddDisk.name)).toBe(false);
+    },
+    VM_ACTIONS_TIMEOUT_SECS * 2,
+  );
+
+  xit(
+    'BZ(1732598) Add/remove nic on VM Network Interfaces page',
+    async () => {
+      await vm.addNIC(networkInterface);
+
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        false,
+      );
+
+      await vm.action('Restart');
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        true,
+      );
+
+      await vm.removeNIC(networkInterface.name);
+      expect((await vm.getAttachedNICs()).includes(networkInterface)).toBe(false);
+
+      await vm.action('Restart');
+
+      expect(searchYAML(networkInterface.networkDefinition, vm.name, vm.namespace, 'vmi')).toBe(
+        false,
+      );
+    },
+    VM_ACTIONS_TIMEOUT_SECS * 2,
+  );
+
+  it('NIC cannot be added twice using one net-attach-def', async () => {
+    await vm.navigateToTab(TABS.NICS);
+    if (
+      (await vm.getAttachedNICs()).filter((nic) => nic.name === networkInterface.name).length === 0
+    ) {
+      await vm.addNIC(networkInterface);
+    }
+
+    // Verify the NIC is added in VM Manifest
+    const resource = getResourceObject(vm.name, vm.namespace, vm.kind);
+    const nic = _.find(
+      _.get(resource, 'spec.template.spec.domain.devices.interfaces', []),
+      (o) => o.name === networkInterface.name,
+    );
+    expect(nic).not.toBe(undefined);
+
+    // Try to add the NIC again
+    await click(createNic, 1000);
+    await browser.sleep(1000).then(() => browser.wait(until.presenceOf($(networkTypeDropdownId))));
```

Comment:
I have hard time to understand this flow without spending time on trying it. Can you please _briefly_ describe these 2 lines (like what is the starting point and what is being tested here) ?

Reply:
We test that it's not possible to add 2 NICs that use the same network (net-attach-def)